app/controllers/assignments_controller.rb

```diff
@@ -138,4 +146,43 @@ def start_upload(assignment, file)
       # Launch java program to process submissions
       SubmissionsHandler.process_submissions(submissions_path, assignment)
   end
+
+  # Responsible for verifying whether a uploaded file is zip by checking its mime type and/or whether can it be extracted by the zip library.
+  # For files with mime type = application/octet-stream, it needs to be further verified by the zip library as it can be a rar file.
+  # Params:
+  # +mimeType+:: string that contains the file's mimetype
+  # +filePath+:: string that contains the file's path which is to be used by the zip library when extracting the file
+  def is_valid_zip?(mimeType, filePath)
+    # Valid zip file mime types that does not required to be further verified by the zip library
+    if mimeType == X_ZIP_COMPRESSED_MIME_TYPE || 
+      mimeType == ZIP_COMPRESSED_MIME_TYPE ||
+      mimeType == APPLICATION_ZIP_MIME_TYPE ||
+      mimeType == MULTIPART_X_ZIP_MIME_TYPE
+      return true;
+    # Need to be further verified by zip library as it can be a rar file
+    elsif if mimeType == OCTET_STREAM_MIME_TYPE && is_opened_as_zip?(filePath)
```

Comment:
I realized we only need to use the rubyzip library for files with mime type, application/octet-stream, as these files can be rar or zip files. The other mime types cover all the zip files